python/ray/worker.py

```diff
@@ -927,25 +928,81 @@ def print_to_stdstream(data):
     print_worker_logs(data, print_file)
 
 
+# Start time of this process, used for relative time logs.
+t0 = time.time()
+autoscaler_log_fyi_printed = False
+
+
+def filter_autoscaler_events(lines):
+    global autoscaler_log_fyi_printed
+
+    if not AUTOSCALER_EVENTS:
+        return
+
+    # Print out autoscaler events only, ignoring other messages.
+    for line in lines:
+        if ":event_summary:" in line:
+            if not autoscaler_log_fyi_printed:
+                yield ("Tip: use `ray status` to view detailed "
+                       "autoscaling status. To disable autoscaler event "
+                       "messages, you can set AUTOSCALER_EVENTS=0.")
+                autoscaler_log_fyi_printed = True
+            yield line.split(":event_summary:")[1]
+
+
+def time_string():
+    delta = time.time() - t0
+    hours = 0
+    minutes = 0
+    while delta > 3600:
+        hours += 1
+        delta -= 3600
+    while delta > 60:
+        minutes += 1
+        delta -= 60
+    output = ""
+    if hours:
+        output += "{}h".format(hours)
+    if minutes:
+        output += "{}m".format(minutes)
```

Comment:
```
hours = int(delta/3600)
if hours > 0:
    output +="{}h".format(hours)
delta = delta-hours*3600
minutes = int(delta/60)
if minutes:
    output += "{}m".format(minutes)
delta = delta-minutes*60
output += "{}s".format(int(delta))
```

Reply:
It's not worth this optimization. Actually, aside from the hours, the rest of the time is already O(60) = O(1).

Reply:
I don't understand why it is not worth this optimization? What harm will it make?

Reply:
It's not worth the readability cost here, and makes no performance difference in practice